lxd/daemon.go

```diff
@@ -388,7 +388,8 @@ func (d *Daemon) Authenticate(w http.ResponseWriter, r *http.Request) (trusted b
 	}
 
 	// Devlxd unix socket credentials on main API.
-	if r.RemoteAddr == devlxdRemoteAddress {
+	authMethod, _ := auth.GetAuthenticationMethodFromCtx(r.Context())
+	if authMethod == api.AuthenticationMethodDevLxd {
```

Comment:
I wonder if it might be worth adding something like:
```golang
// lxd/auth/util.go
func IsAuthenticationMethod(ctx context.Context, method string) (bool, error) {
    actualMethod, err := GetAuthenticationMethodFromCtx(ctx)
    if err != nil {
        return false, err
    }

    return method == actualMethod
}
```

Reply:
I was thinking the same, but based on the current usage I don't think this does much in making code more readable nor shorter.